src/client/gameui.cpp

```diff
@@ -243,8 +243,14 @@ void GameUI::setChatText(const EnrichedString &chat_text, u32 recent_chat_count)
 	chat_size.LowerRightCorner.Y = std::min((s32)window_size.Y,
 		m_guitext_chat->getTextHeight() + chat_y);
 
-	m_guitext_chat->setRelativePosition(chat_size);
-	setStaticText(m_guitext_chat, chat_text);
+	if (chat_size != m_cached_chat_size) {
+		m_guitext_chat->setRelativePosition(chat_size);
+		m_cached_chat_size = chat_size;
+	}
+	if (chat_text != m_cached_chat_text) {
+		setStaticText(m_guitext_chat, chat_text);
+		m_cached_chat_text = chat_text;
+	}
 
 	m_recent_chat_count = recent_chat_count;
 }
```

Comment:
What about instead of changing `Game::updateChat()` to not call this function when nothing changed?
Seems like the better approach and  possible to me with a bit of rework in `ChatBackend` (implement marking as modified).

It would also avoid doing *this* every frame:
![grafik](https://user-images.githubusercontent.com/1042418/132996736-51e47749-f2fe-490b-a534-c984a8c61678.png)

Reply:
:+1:  Done.